### PR TITLE
Adding more EVE displays for extui

### PR DESCRIPTION
**Pull request by RudolphRiedel**
### Requirements
Extui GUI and Matrix Orbital EVE2-50G or EVE3-50G with adapter-board.

### Description
I started to add more displays to extui now that my display adapter boards are up and running:
https://github.com/RudolphRiedel/EVE_display-adapter

Work in progress with more to follow.
### Benefits
Adds more displays that can be used with Marlin.

### CODE REVIEW

**Comment by RudolphRiedel** [comment]
I am confused, am I supposed to do anything here?
In any case I am kind of blocked out for over a week now.

And about the changes you did:
>#if ENABLED(TOUCH_UI_DEBUG)
>   if (counter > 248) {

to

>if (ENABLED(TOUCH_UI_DEBUG) && counter > 248) {

Doesn't your version generate code even if TOUCH_UI_DEBUG is not set?
At the very least syntax highlighting in vscode does not work this way, the lines are shown as active even thogh
TOUCH_UI_DEBUG is set to false.

Excuse my ignorance, but how is that an improvement?

**Comment by RudolphRiedel** [comment]
@thinkyhead What is wrong with this pull request?

File: Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h [hunk 1]
```diff
@@ -177,8 +177,105 @@
...
 #else
 
   #error "Unknown or no TOUCH_UI_FTDI_EVE board specified. To add a new board, modify this file."
 
 #endif
+
+
+/* this data is used to patch FT813 displays that use a GT911 as a touch-controller */
+#ifdef PATCH_GT911
+  constexpr PROGMEM unsigned char GT911_data[] = {
```
Comment (sjasonsmith) [inline Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h:254]: What does this do? Where did it com from?

Comment (RudolphRiedel) [reply Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h:254]: This is from Bridgeteks application note "AN_336 FT8xx - Selecting an LCD Display" and it applies a patch to the FT813 in order to make it support Goodix GT911 touch controllers.
Out of the box the FT813 mostly supports FocalTech touch controllers, the IQS550 from Azoteq and the HY4613 from Hycon.

The EVE2-xxG modules from Matrix Orbital need this patch and the Crystalfontz CFAF800480E0-050.
BT815 like in the EVE3 series do support the GT911 natively but have to configured for it.

This is why this is inside an #ifdef, to usually not add this to the binary.

Comment (sjasonsmith) [reply Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h:254]: Would you mind adding this or a similar explanation to the comment in the code? I can imagine someone looking at it later and asking the same questions I did.

I looked up that AN you mentioned. I must say I was a little surprised to find big binary blobs like that in a PDF :)

**Comment by sjasonsmith** [comment]
> What is wrong with this pull request?

@RudolphRiedel if you think this is ready to go in, you should mark it Ready For Review so it no longer shows up as Draft.

File: Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h (hunk 1 above)
Comment (RudolphRiedel) [reply Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/basic/boards.h:254]: Well, there are comments in to explain what this is, not only line 252 but also lines 199 and 231.
And line 1102ff of commands.pp

**Comment by RudolphRiedel** [comment]
I did not put it ony draft, @thinkyhead did after he started to mess around with it for no apparent reason.
And instead of doing anything else he chose to ignore me.

Not to mention that when I put this in it did not have any conflicts and was tested.

**Comment by sjasonsmith** [comment]
Oh, I didn't notice the merge conflict. At this point I'm sure the whole thing has to be rebased. I understand the frustration.

**Comment by thinkyhead** [comment]
Thanks!

**Pull request merged**